app/models/product_test.rb

```diff
@@ -108,10 +108,9 @@ def sample_and_duplicate_records(recs, ids, random: Random.new)
     dups = records.find(ids)
 
     recs, dups = randomize_clinical_data(recs, dups, random)
-
-    (random.rand(3) + 1).times do
+    dups.sample(random.rand(3) + 1, random: random).each do |rec|
```

Comment:
My mistake you are correct @okeefm. 
So yeah @ljades like he said we want `random.rand(1..3)`